src/sql/src/pure.rs

```diff
@@ -382,13 +363,22 @@ async fn purify_csr_connector_proto(
     };
 
     let CsrConnectorProto {
-        url,
+        connector,
         seed,
         with_options: ccsr_options,
     } = csr_connector;
     match seed {
         None => {
-            let url: Url = url.parse()?;
+            let url: Url = match connector {
+                CsrConnector::Inline { url: uri } => uri.parse()?,
+                CsrConnector::Reference { url, .. } => match url {
```

Comment:
think this should be `expect`

Reply:
When I tried using expect here I ran into
```
cannot move out of `*url` which is behind a mutable reference
move occurs because `*url` has type `std::option::Option<std::string::String>`, which does not implement the `Copy` trait
```
and adding `as_ref()` as suggested then breaks type inference.

Reply:
Hmm, I'm not able to duplicate that issue at all, and this code works for me w/o issue:

```diff
diff --git a/src/sql/src/pure.rs b/src/sql/src/pure.rs
index e95253637..357b8728e 100644
--- a/src/sql/src/pure.rs
+++ b/src/sql/src/pure.rs
@@ -23,7 +23,6 @@ use mz_sql_parser::ast::{CsrConnector, KafkaConnector, KafkaSourceConnector};
 use prost::Message;
 use protobuf_native::compiler::{SourceTreeDescriptorDatabase, VirtualSourceTree};
 use protobuf_native::MessageLite;
-use reqwest::Url;
 use tokio::fs::File;
 use tokio::io::AsyncBufReadExt;
 use tokio::task;
@@ -366,13 +365,11 @@ async fn purify_csr_connector_proto(
     } = csr_connector;
     match seed {
         None => {
-            let url: Url = match connector {
-                CsrConnector::Inline { url: uri } => uri.parse()?,
-                CsrConnector::Reference { url, .. } => match url {
-                    Some(url) => url.parse()?,
-                    None => bail!("CSR Connector must specify a url!"),
-                },
-            };
+            let url = match connector {
+                CsrConnector::Inline { url } => url,
+                CsrConnector::Reference { url, .. } => url.as_ref().unwrap(),
+            }
+            .parse()?;
             let kafka_options = kafka_util::extract_config(&mut normalize::options(with_options))?;
             let ccsr_config = kafka_util::generate_ccsr_client_config(
                 url,

```

Let me know if you'd like to chat about how `Option`/`unwrap`/`as_ref` interact. Certainly not the most intuitive paradigm to work within.

Reply:
I was failing to notice moving the `parse()?` to the end of the match block which lets me use `unwrap()` or `expect()` as you suggest.